### PR TITLE
CANCEL SESSION / CANCEL QUERY / SHOW SESSIONS

### LINKED ISSUE #2047: Transaction / session cancellation
Background: https://airtable.com/tblD3oZPLJgGhCmch/viwINAgXA5on9pr2T/recGdKHLMwjLLbJ1C
PM: @dianasaur323 
Eng Partner: @jordanlewis

### PR DESCRIPTION
**Pull request by lhirata**
Closes #2047.

### CODE REVIEW

**Comment by lhirata** [comment]
@jordanlewis - when I regenerated the SQL diagrams for CANCEL QUERY / CANCEL SESSION, I noticed that "CANCEL QUERIES" and "CANCEL SESSIONS" show up. It seems like they don't work though. Is this expected?

```
CANCEL SESSIONS (SELECT session_id FROM [SHOW SESSIONS]                                                                                                                                        WHERE username = 'maxroach');
invalid syntax: statement ignored: syntax error at or near "sessions"
DETAIL: source SQL:
CANCEL SESSIONS (SELECT session_id FROM [SHOW SESSIONS]
       ^
```

**Comment by jordanlewis** [comment]
:lgtm: although like we said offline I think `CANCEL SESSIONS` should be working.

---

Review status: 0 of 6 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[v2.1/cancel-session.md, line 16 at r1](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD83PFe8iHBnS5meQWX:-LD83PFe8iHBnS5meQWY:b-78sf5o) ([raw file](https://github.com/cockroachdb/docs/blob/859f62363269ecb48bfe8987354f4d68ed97ae6b/v2.1/cancel-session.md#L16)):*
> ```Markdown
> ## Required Privileges
> 
> By default, only the `root` user can cancel a session.
> ```

correction: it's only the `root` user and the user that the session belongs to. Also, this is not just by default - it's always, because it's not currently configurable.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181#-:-LD83ovI9oqzi97xOIxJ:bk2o2t1)*

File: v2.1/cancel-query.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ The `root` user can cancel any currently active queries, whereas non-`root` user
 Parameter | Description
 ----------|------------
 `query_id` | A [scalar expression](scalar-expressions.html) that produces the ID of the query to cancel.<br><br>`CANCEL QUERY` accepts a single query ID. If a subquery is used and returns multiple IDs, the `CANCEL QUERY` statement will therefore fail.
```
Comment (rmloveland) [inline v2.1/cancel-query.md]: I would go with "the ... statement will fail", the "therefore" is unnecessary IMO.

Comment (rmloveland) [reply v2.1/cancel-query.md]: It seems like this should be updated with the same text as the note below, re: using `CANCEL QUERIES` in the case where there are multiple query IDs returned.  Or am I misunderstanding what I'm seeing?

File: v2.1/cancel-query.md [hunk 2]
```diff
@@ -69,7 +70,7 @@ In this example, we nest a [`SELECT` clause](select-clause.html) that retrieves
           AND query = 'SELECT * FROM test.kv ORDER BY k');
 ~~~
 
-{{site.data.alerts.callout_info}}<code>CANCEL QUERY</code> accepts a single query ID. If subquery is used and returns multiple IDs, the <code>CANCEL QUERY</code> statement will therefore fail.{{site.data.alerts.end}}
+{{site.data.alerts.callout_info}}<code>CANCEL QUERY</code> accepts a single query ID. If a subquery is used and returns multiple IDs, the <code>CANCEL QUERY</code> statement will fail. To cancel multiple queries, use the <code>CANCEL QUERIES</code> syntax.{{site.data.alerts.end}}
```
Comment (rmloveland) [inline v2.1/cancel-query.md]: I might simplify to the shorter "To cancel multiple queries, use CANCEL QUERIES".

File: v2.1/cancel-session.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+summary: The CANCEL SESSION statement stops long-running sessions.
+toc: false
+---
+
+<span class="version-tag">New in v2.1:</span> The `CANCEL SESSION` [statement](sql-statements.html) lets you stop long-running sessions.
+
+<div id="toc"></div>
+
+## Considerations
+
```
Comment (rmloveland) [inline v2.1/cancel-session.md]: blank heading - possibly left in by accident?

File: v2.1/cancel-session.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+In this example, we nest a [`SELECT` clause](select-clause.html) that retrieves the ID of a session inside the `CANCEL SESSION` statement:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CANCEL SESSIONS (SELECT session_id FROM [SHOW SESSIONS]
+      WHERE username = 'maxroach');
+~~~
+
+{{site.data.alerts.callout_info}}<code>CANCEL SESSION</code> accepts a single query ID. If a subquery is used and returns multiple IDs, the <code>CANCEL SESSIONS</code> statement will fail. To cancel multiple sessions, use the <code>CANCEL SESSIONS</code> syntax.{{site.data.alerts.end}}
```
Comment (rmloveland) [inline v2.1/cancel-session.md]: Same comment as above re: "use CANCEL SESSIONS"

File: v2.1/cancel-session.md (hunk 3 above)
Comment (rmloveland) [reply v2.1/cancel-session.md]: Following up on this - what might the consequences be of cancelling a session?  E.g., if you have a session that's in the middle of doing a lot of things and you cancel the session can you wedge the cluster?  (Presumably the answer is no, but perhaps there are scenarios when it is non-optimal?)

Sorry if this is not a copy-edit question!

**Comment by lhirata** [comment]
Review status: 0 of 7 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-query.md, line 73 at r3](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD7Yhsgr3044AOlGkzg-r3-73:-LDDCFzI7JPMjTdlKhYq:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/dce37894278b635ed86f7e7a4452e069496ae456/v2.1/cancel-query.md#L73)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

General comment: should this page link to CANCEL SESSION at the bottom?  Seems related.
</blockquote></details>

Done.

---

*[v2.1/cancel-session.md, line 16 at r1](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD83PFe8iHBnS5meQWX:-LDDDPle3gollwTaNel_:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/859f62363269ecb48bfe8987354f4d68ed97ae6b/v2.1/cancel-session.md#L16)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

correction: it's only the `root` user and the user that the session belongs to. Also, this is not just by default - it's always, because it's not currently configurable.
</blockquote></details>

Done.

---

*[v2.1/cancel-session.md, line 12 at r3](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD7Yhsgr3044AOlGkzj-r3-12:-LDDCwDb42OPsttKQPQf:b-d6ncdp) ([raw file](https://github.com/cockroachdb/docs/blob/dce37894278b635ed86f7e7a4452e069496ae456/v2.1/cancel-session.md#L12)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Following up on this - what might the consequences be of cancelling a session?  E.g., if you have a session that's in the middle of doing a lot of things and you cancel the session can you wedge the cluster?  (Presumably the answer is no, but perhaps there are scenarios when it is non-optimal?)

Sorry if this is not a copy-edit question!
</blockquote></details>

Good point, Rich!

@jordanlewis, should I add any considerations for CANCEL SESSION? What happens if you CANCEL SESSION and it's in the middle of something?

---

*[v2.1/cancel-session.md, line 64 at r3](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD7Yhsgr3044AOlGkzj-r3-64:-LDDDUkc2Y2knds_Dyit:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/dce37894278b635ed86f7e7a4452e069496ae456/v2.1/cancel-session.md#L64)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Same comment as above re: "use CANCEL SESSIONS"
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181)*

**Comment by jseldess** [comment]
Just a few comments and questions, @lhirata. 

Also wondering if we should update this page: https://www.cockroachlabs.com/docs/dev/manage-long-running-queries.html

---

Review status: 0 of 7 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-session.md, line 24 at r6](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDI-DFuD0mtuu4p2Vd8:-LDI-DFvBjkR_UTJ7xs_:bv624vw) ([raw file](https://github.com/cockroachdb/docs/blob/9960734d0014f1d630f367624af21537ab82c486/v2.1/cancel-session.md#L24)):*
> ```Markdown
> ----------|------------
> `session_id` | The ID of the session you want to cancel, which can be found with [`SHOW SESSIONS`](show-sessions.html).<br><br>`CANCEL SESSION` accepts a single session ID. If a subquery is used and returns multiple IDs, the `CANCEL SESSION` statement will fail. To cancel multiple sessions, use `CANCEL SESSIONS`.
> `select_stmt` | A [selection query](selection-queries.html) whose result you want to cancel.
> ```

I think we should make this a bit more specific, clarifying that the "result" of this query must be the ID of one or more sessions that you want to cancel? Not sure of the best formulation

---

*[v2.1/cancel-session.md, line 28 at r6](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDI-Uq89sFe8Fs_XDYL:-LDI-Uq89sFe8Fs_XDYM:brx4zjp) ([raw file](https://github.com/cockroachdb/docs/blob/9960734d0014f1d630f367624af21537ab82c486/v2.1/cancel-session.md#L28)):*
> ```Markdown
> ## Example
> 
> ### Cancel a Query via the Session ID
> ```

I think it might make more sense to frame the 2 examples as `Cancel a Single Session` and `Cancel Multiple Sessions`. 

For the former, you could show both cancelling by the session ID and by a select query that returns only a single session ID. For the latter, your basic example is fine but you could clarify that all session IDs returned by the select query would be cancelled. 

---

*[v2.1/cancel-session.md, line 53 at r6](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDI01n8A4NrceVzl_O-:-LDI01n8A4NrceVzl_O0:b-u0b3p9) ([raw file](https://github.com/cockroachdb/docs/blob/9960734d0014f1d630f367624af21537ab82c486/v2.1/cancel-session.md#L53)):*
> ```Markdown
> ### Cancel a Session via a Subquery
> 
> In this example, we nest a [`SELECT` clause](select-clause.html) that retrieves the ID of a session inside the `CANCEL SESSION` statement:
> ```

See comment above: I think it's important to use this example to show how to cancel multiple sessions. In this case, it would be all sessions created by the `maxroach` users.

---

*[v2.1/sql-statements.md, line 120 at r6](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDI0rGG6WlniygNw3LG:-LDI0rGG6WlniygNw3LH:bz6x3cp) ([raw file](https://github.com/cockroachdb/docs/blob/9960734d0014f1d630f367624af21537ab82c486/v2.1/sql-statements.md#L120)):*
> ```Markdown
> [`SHOW ALL CLUSTER SETTINGS`](show-cluster-setting.html) | List the current cluster-wide settings.
> [`SHOW SESSIONS`](show-sessions.html) | List details about currently active sessions.
> [`CANCEL SESSION`](cancel-session.html) | Cancel a long-running session.
> ```

Please add the "New in 2.1" callout.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181#-:-LDI0y9q0Z3oJAR1gp4p:b-laydko)*

**Comment by lhirata** [comment]
Review status: 0 of 7 files reviewed at latest revision, 9 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-query.md, line 28 at r3](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD7Yhsgr3044AOlGkzg-r3-28:-LDI59NrEcAOXWYvPdIs:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/dce37894278b635ed86f7e7a4452e069496ae456/v2.1/cancel-query.md#L28)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

It seems like this should be updated with the same text as the note below, re: using `CANCEL QUERIES` in the case where there are multiple query IDs returned.  Or am I misunderstanding what I'm seeing?
</blockquote></details>

Done.

---

*[v2.1/cancel-session.md, line 24 at r6](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDI-DFuD0mtuu4p2Vd8:-LDI8mfg6eRTf7QAvJmm:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9960734d0014f1d630f367624af21537ab82c486/v2.1/cancel-session.md#L24)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think we should make this a bit more specific, clarifying that the "result" of this query must be the ID of one or more sessions that you want to cancel? Not sure of the best formulation
</blockquote></details>

Done.

---

*[v2.1/cancel-session.md, line 28 at r6](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDI-Uq89sFe8Fs_XDYL:-LDI5Qoe6mlhEd7HENA8:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9960734d0014f1d630f367624af21537ab82c486/v2.1/cancel-session.md#L28)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think it might make more sense to frame the 2 examples as `Cancel a Single Session` and `Cancel Multiple Sessions`. 

For the former, you could show both cancelling by the session ID and by a select query that returns only a single session ID. For the latter, your basic example is fine but you could clarify that all session IDs returned by the select query would be cancelled. 
</blockquote></details>

Done.

---

*[v2.1/cancel-session.md, line 53 at r6](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDI01n8A4NrceVzl_O-:-LDI7KdJ8KICoJBk8ctx:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9960734d0014f1d630f367624af21537ab82c486/v2.1/cancel-session.md#L53)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

See comment above: I think it's important to use this example to show how to cancel multiple sessions. In this case, it would be all sessions created by the `maxroach` users.
</blockquote></details>

Done.

---

*[v2.1/sql-statements.md, line 120 at r6](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDI0rGG6WlniygNw3LG:-LDI7YJRAr3kwNYLeOU4:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9960734d0014f1d630f367624af21537ab82c486/v2.1/sql-statements.md#L120)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Please add the "New in 2.1" callout.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181)*

**Comment by jseldess** [comment]
:lgtm:, pending a few nits and a response from Jordan.

---

Review status: 0 of 7 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-session.md, line 12 at r3](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD7Yhsgr3044AOlGkzj-r3-12:-LDICguOEKaEVnrEW9_i:bpkm3yh) ([raw file](https://github.com/cockroachdb/docs/blob/dce37894278b635ed86f7e7a4452e069496ae456/v2.1/cancel-session.md#L12)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

Good point, Rich!

@jordanlewis, should I add any considerations for CANCEL SESSION? What happens if you CANCEL SESSION and it's in the middle of something?
</blockquote></details>

@jordanlewis, I think these details are important. Are there cases where cancelling a session will NOT cancel the in-progress queries that the session initiated? 

---

*[v2.1/cancel-session.md, line 79 at r7](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDICBMGBvcPxaCF9YEJ:-LDICBMGBvcPxaCF9YEK:buktr92) ([raw file](https://github.com/cockroachdb/docs/blob/df7f7a3858fd155aa8f4343ccd3557ccd7ee7a97/v2.1/cancel-session.md#L79)):*
> ```Markdown
> ~~~
> 
>  To cancel multiple sessions, nest a [`SELECT` clause](select-clause.html) that retrieves `session_id`(s) inside the `CANCEL SESSION` statement:
> ```

In the intro, `CANCEL SESSION` should be `CANCEL SESSIONS`.

---

*[v2.1/cancel-session.md, line 87 at r7](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDICUAMBnd_Pg6YidJZ:-LDICUAMBnd_Pg6YidJ_:bavw4w2) ([raw file](https://github.com/cockroachdb/docs/blob/df7f7a3858fd155aa8f4343ccd3557ccd7ee7a97/v2.1/cancel-session.md#L87)):*
> ```Markdown
> ~~~
> 
> All sessions created by 'maxroach' will be cancelled.
> ```

nit: probably fine to just put maxroach in backticks rather than single quotes.

---

*[v2.1/cancel-session.md, line 89 at r7](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDIC2lF82OyhiEKe5T8:-LDIC2lF82OyhiEKe5T9:bvcs8bj) ([raw file](https://github.com/cockroachdb/docs/blob/df7f7a3858fd155aa8f4343ccd3557ccd7ee7a97/v2.1/cancel-session.md#L89)):*
> ```Markdown
> All sessions created by 'maxroach' will be cancelled.
> 
> {{site.data.alerts.callout_info}}<code>CANCEL SESSION</code> accepts a single session ID. If a subquery is used and returns multiple IDs, the <code>CANCEL SESSION</code> statement will fail. To cancel multiple sessions, use <code>CANCEL SESSIONS</code>.{{site.data.alerts.end}}
> ```

I think this callout would make more sense at the end of the `Cancel a Single Session` example. In this `Cancel Multiple Sessions` example, you would never use the singular `CANCEL SESSION`.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181#-:-LDID2JUAFeTTTELLleq:b-d36w12)*

**Comment by lhirata** [comment]
Review status: 0 of 7 files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-session.md, line 79 at r7](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDICBMGBvcPxaCF9YEJ:-LDInaZVEyJxCGWWuIBk:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/df7f7a3858fd155aa8f4343ccd3557ccd7ee7a97/v2.1/cancel-session.md#L79)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

In the intro, `CANCEL SESSION` should be `CANCEL SESSIONS`.
</blockquote></details>

Done.

---

*[v2.1/cancel-session.md, line 87 at r7](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDICUAMBnd_Pg6YidJZ:-LDIne8IB349CmaBvnSP:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/df7f7a3858fd155aa8f4343ccd3557ccd7ee7a97/v2.1/cancel-session.md#L87)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: probably fine to just put maxroach in backticks rather than single quotes.
</blockquote></details>

Done.

---

*[v2.1/cancel-session.md, line 89 at r7](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDIC2lF82OyhiEKe5T8:-LDIngnd-uz5iXvNs2YI:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/df7f7a3858fd155aa8f4343ccd3557ccd7ee7a97/v2.1/cancel-session.md#L89)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think this callout would make more sense at the end of the `Cancel a Single Session` example. In this `Cancel Multiple Sessions` example, you would never use the singular `CANCEL SESSION`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181)*

**Comment by jordanlewis** [comment]
:lgtm:!

---

Reviewed 4 of 6 files at r1, 1 of 2 files at r4, 1 of 2 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-session.md, line 12 at r3](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD7Yhsgr3044AOlGkzj-r3-12:-LDMhJmy12ubWvWs7M6x:b5s5p02) ([raw file](https://github.com/cockroachdb/docs/blob/dce37894278b635ed86f7e7a4452e069496ae456/v2.1/cancel-session.md#L12)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

@jordanlewis, I think these details are important. Are there cases where cancelling a session will NOT cancel the in-progress queries that the session initiated? 
</blockquote></details>

CANCEL SESSION will always try to cancel the session's in-progress query. It may not succeed, but it will always attempt to do so. There aren't really any interesting edge cases, as I understand, though. I think the wording is currently accurate!

---

*[v2.1/cancel-session.md, line 56 at r8](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDMhy0K3fibfjEYv5nB:-LDMhy0K3fibfjEYv5nC:b-6dotd5) ([raw file](https://github.com/cockroachdb/docs/blob/746a6f69468df1630ae9330f26188d048c999ff7/v2.1/cancel-session.md#L56)):*
> ```Markdown
> ~~~ sql
> > CANCEL SESSIONS (SELECT session_id FROM [SHOW SESSIONS]
>       WHERE username = 'root');
> ```

Does it make sense to have this `CANCEL SESSIONS` example in the `CANCEL SESSION` section?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181#-:-LDMjkzh1xyQHhG4uDh4:bg7igkw)*

**Comment by jseldess** [comment]
*[v2.1/cancel-session.md, line 12 at r3](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD7Yhsgr3044AOlGkzj-r3-12:-LDMmmKv6GISQ_uIG8lt:b-8x8yy9) ([raw file](https://github.com/cockroachdb/docs/blob/dce37894278b635ed86f7e7a4452e069496ae456/v2.1/cancel-session.md#L12)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

CANCEL SESSION will always try to cancel the session's in-progress query. It may not succeed, but it will always attempt to do so. There aren't really any interesting edge cases, as I understand, though. I think the wording is currently accurate!
</blockquote></details>

Thanks, Jordan. @lhirata, your call about whether to soften or somehow alter such statements like `will cancel the currently active query and end the session` in the intro.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181)*

**Comment by lhirata** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-session.md, line 56 at r8](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDMhy0K3fibfjEYv5nB:-LDMvEXsDyaV8A2bPsr-:bmiv2w6) ([raw file](https://github.com/cockroachdb/docs/blob/746a6f69468df1630ae9330f26188d048c999ff7/v2.1/cancel-session.md#L56)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

Does it make sense to have this `CANCEL SESSIONS` example in the `CANCEL SESSION` section?
</blockquote></details>

I wanted to show that you could cancel one session if only one session is returned using a subquery, but I can see that the syntax is confusing. It seems like CANCEL SESSION (subquery) works, but the way the SQL diagram is laid out seems to suggest you should only use a subquery if you use the plural version of the statement. 

Should I just remove this example?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181)*

**Comment by jordanlewis** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-session.md, line 56 at r8](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDMhy0K3fibfjEYv5nB:-LDN-07T1DB-UheLoeHs:b-m078sj) ([raw file](https://github.com/cockroachdb/docs/blob/746a6f69468df1630ae9330f26188d048c999ff7/v2.1/cancel-session.md#L56)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

I wanted to show that you could cancel one session if only one session is returned using a subquery, but I can see that the syntax is confusing. It seems like CANCEL SESSION (subquery) works, but the way the SQL diagram is laid out seems to suggest you should only use a subquery if you use the plural version of the statement. 

Should I just remove this example?
</blockquote></details>

No, I think it's a great example! I just noticed that it was in the single session area which felt inconsistent - but I'll definitely defer to you on this.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181)*

**Comment by lhirata** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[v2.1/cancel-session.md, line 12 at r3](https://reviewable.io/reviews/cockroachdb/docs/3181#-LD7Yhsgr3044AOlGkzj-r3-12:-LDgNUxAAzgHfJCPBv_m:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/dce37894278b635ed86f7e7a4452e069496ae456/v2.1/cancel-session.md#L12)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Thanks, Jordan. @lhirata, your call about whether to soften or somehow alter such statements like `will cancel the currently active query and end the session` in the intro.
</blockquote></details>

Done.

---

*[v2.1/cancel-session.md, line 56 at r8](https://reviewable.io/reviews/cockroachdb/docs/3181#-LDMhy0K3fibfjEYv5nB:-LDgNgoL1m_Z2FP2Uj-7:bkfvlgq) ([raw file](https://github.com/cockroachdb/docs/blob/746a6f69468df1630ae9330f26188d048c999ff7/v2.1/cancel-session.md#L56)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

No, I think it's a great example! I just noticed that it was in the single session area which felt inconsistent - but I'll definitely defer to you on this.
</blockquote></details>

Ok, leaving as is!

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3181)*

**Pull request merged**